### DISCUSSION TITLE
Can't connect to a website hosted on WSL2 from host machine using mirrored networking but can from another device on LAN

### DISCUSSION THREAD

**Discussion by bbceg** [microsoft/WSL#11944] [Q&A]
Earlier today I switched from WSL1 to WSL2. As the title suggests, I can't connect to a website running on WSL2 from the host machine despite being able to connect from my phone without issue. I am however able to connect to the WSL2 instance with VSCode. I presume this is some kind of firewall problem but I'm not sure where to start...

Here's my .wslconfig:
```
[wsl2]
networkingMode=mirrored
dnsTunneling=true
localhostForwarding=true
```
And here are the commands I've tried as specified in the docs (I gather only one is required but I tried both for good measure!):
```
Set-NetFirewallHyperVVMSetting -Name '{40E0AC32-46A5-438A-A0B2-2B479E8F2E90}' -DefaultInboundAction Allow

New-NetFirewallHyperVRule -Name "MyWebServer" -DisplayName "My Web Server" -Direction Inbound -VMCreatorId '{40E0AC32-46A5-438A-A0B2-2B479E8F2E90}' -Protocol TCP -LocalPorts 80
```

I need to be able to access the websites on WSL2 from any machine on the LAN (including my own, which is the host). I have a local DNS server which allows machines on the LAN to find the host with any .localdev domain lookup, removing the need for a host file entry for each website on every machine.

Hope someone can help!

**Comment by bbceg** [comment]
Just to clarify, if I add an entry for each website to my hosts file with 127.0.0.1 on the host machine everything works as expected. It's only when I try to use the LAN IP of the host machine that it fails, even though this is what all other machines on the network are using...

**Comment by golavr** [comment]
@bbceg Have you managed to solve it?
I'm getting the same behavior as you described.

**Reply by bbceg** [reply]
After battling away with it again this morning I'm afraid I got no further. Instead I'm using [Acrylic DNS Proxy
](https://mayakron.altervista.org/support/acrylic/Home.htm) to allow a wildcard hosts file. Please do let me know if you ever manage to fix it.

**Reply by bbceg** [reply]
@golavr turns out this wasn't a good idea for my use case. I spent most of the day today trying to work out why an ajax call wouldn't work, only to find it was this change. So I'm now back to defining individual entries for all local websites in my windows hosts file.
